direct/core/src/main/java/cz/o2/proxima/direct/view/LocalCachedPartitionedView.java

```diff
@@ -313,13 +313,12 @@ private void scanWildcardPrefix(
   @Override
   public void listEntities(
       RandomOffset offset, int limit, Consumer<Pair<RandomOffset, String>> consumer) {
-
-    IntOffset off = (IntOffset) offset;
-    AtomicInteger newOff = new AtomicInteger(off.getOffset());
+    final IntOffset off = offset == null ? new IntOffset(0) : (IntOffset) offset;
+    final AtomicInteger newOff = new AtomicInteger(off.getOffset());
     cache.keys(
         off.getOffset(),
         limit,
-        key -> consumer.accept(Pair.of(new IntOffset(newOff.incrementAndGet()), key)));
+        key -> consumer.accept(Pair.of(new IntOffset(newOff.getAndIncrement()), key)));
```

Comment:
I'm not sure about this. It seems to pass tests, but that probably means the tests are broken. :-)
An invariant should be that:
```java
  Offset firstOffset = view.fetchOffset(Listing.ENTITY, "");
  List<Offset> next = new ArrayList<>();
  view.listEntities(firstOffset, 1, p -> next.add(p.getFirst());
  assertNotEquals(first, next.get(0));
```

Reply:
So the first record should start on index 1? That seems kind of non-standard :)

Reply:
That's not the index of first record. That is index of first record that follows the record you just received.